packages/core/src/codewhisperer/service/transformationResultsViewProvider.ts

```diff
@@ -330,18 +326,26 @@ export class ProposedTransformationExplorer {
                     'gumby.reviewState',
                     TransformByQReviewStatus.NotStarted
                 )
-                const errorMessage = CodeWhispererConstants.errorDownloadingDiffMessage
-                getLogger().error(`CodeTransformation: ExportResultArchive error = ${errorMessage}`)
+                getLogger().error(`CodeTransformation: ExportResultArchive error = ${downloadErrorMessage}`)
                 telemetry.codeTransform_logApiError.emit({
                     codeTransformApiNames: 'ExportResultArchive',
                     codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
                     codeTransformJobId: transformByQState.getJobId(),
-                    codeTransformApiErrorMessage: (e as Error).message ?? errorMessage,
+                    codeTransformApiErrorMessage: downloadErrorMessage,
                     codeTransformRequestId: e.requestId ?? '',
                     result: MetadataResult.Fail,
                     reason: 'ExportResultArchiveFailed',
                 })
                 throw new Error('Error downloading diff')
+            } finally {
+                // This metric is emitted when user clicks Download Proposed Changes button
+                telemetry.codeTransform_vcsViewerClicked.emit({
```

Comment:
Is there a reason for not using `telemetry.codeTransform_vscViewerClicked.run()`?

Reply:
We have work in progress that intends to consolidate all of the custom metrics we've added into one large metric to hold anything/everything we need. I'd say as a part of that work we can transition from the `emit` calls we have everywhere to `run`.

Right above here in the `catch` we have a different metric being emitted, so when we complete that ^ work that metric's info could be captured via `telemetry.consolidated_Metric.record(...)`